src/highlights/v0.24.md

```diff
@@ -0,0 +1,300 @@
+
+We also have a new example demonstrating how to create a custom widget.
+
+![custom widget](https://raw.githubusercontent.com/ratatui-org/ratatui/images/examples/custom_widget.gif)
+
+```sh
+cargo run --example=custom_widget --features=crossterm
+```
+
+> The code is available [here](https://github.com/ratatui-org/ratatui/blob/main/examples/custom_widget.rs).
+
+Lastly, we added an example to demonstrate RGB color options:
+
+![RGB colors](https://raw.githubusercontent.com/ratatui-org/ratatui/images/examples/colors_rgb.gif)
+
+```sh
+cargo run --example=colors_rgb --features=crossterm
+```
+
+> The code is available [here](https://github.com/ratatui-org/ratatui/blob/main/examples/colors_rgb.rs).
+
+## Window Size API 🪟
+
+A new method called `window_size` is added for retrieving the window size.
+It returns a struct called `WindowSize` that contains both pixels (width, height) and columns/rows
+information.
+
+```rs
+let stdout = std::io::stdout();
+let mut backend = CrosstermBackend::new(stdout);
+println!("{:#?}", backend.window_size().unwrap());
+```
+
+Outputs:
+
+```
+WindowSize {
+    columns_rows: Size {
+        width: 240,
+        height: 30,
+    },
+    pixels: Size {
+        width: 0,
+        height: 0,
+    },
+}
+```
+
+With this new API, the goal is to improve image handling in terminal emulators
+by sharing terminal size and layout information,
+enabling precise image placement and resizing within rectangles.
+
+See the pull request for more information:
+[https://github.com/ratatui-org/ratatui/pull/276](https://github.com/ratatui-org/ratatui/pull/276)
+
+## BarChart: Render _smol_ charts 📊
+
+We had a bug where the `BarChart` widget doesn't render labels that are full width.
+Now this is fixed and we are able to render charts smaller than 3 lines!
+
+For example, here is how `BarChart` is rendered and resized from a single line to 4 lines in order:
+
+```
+  ▁ ▂ ▃ ▄ ▅ ▆ ▇ 8
+
+
+  ▁ ▂ ▃ ▄ ▅ ▆ ▇ 8
+a b c d e f g h i
+
+  ▁ ▂ ▃ ▄ ▅ ▆ ▇ 8
+a b c d e f g h i
+      Group
+
+          ▂ ▄ ▆ █
+  ▂ ▄ ▆ 4 5 6 7 8
+a b c d e f g h i
+      Group
+```
+
+If you set `bar_width(2)` then it is rendered as:
+
+```
+          ▂ ▄ ▆ █
+  ▂ ▄ ▆ 4 5 6 7 8
+a b c d e f g h i
+      Group
+```
+
+## Block: custom symbols for borders 🛡️
+
+The `Block` widget has a new method called `border_set` that can be used to specify the symbols that are
+going to be used for the borders.
+
+```rs
+Block::default()
+    .borders(Borders::ALL)
+    .border_set(border::Set {
+        top_left: "1",
+        top_right: "2",
+        bottom_left: "3",
+        bottom_right: "4",
+        vertical_left: "L",
+        vertical_right: "R",
+        horizontal_top: "T",
+        horizontal_bottom: "B",
+    })
+```
+
+When rendered:
+
+```
+1TTTTTTTTTTTTT2
+L             R
+3BBBBBBBBBBBBB4
+```
+
+There are also 2 new border types added (`QuadrantInside`, `QuadrantOutside`).
```

Comment:
We should get a preview of these as an image sometime soon.

Reply:
Yup, I guess it is fine for now.